core/src/main/java/org/apache/iceberg/rest/requests/FetchScanTasksRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface FetchScanTasksRequest extends RESTRequest {
+
+  String planTask();
+
+  @Override
+  default void validate() {
+    Preconditions.checkArgument(
```

Comment:
yeah you won't be able to construct a nullable request since it's a required field in the API itself, thus you can leave `validate()` empty and add a comment saying that it's not possible to create an invalid request (similar to how other immutable request/response classes do it)

Reply:
I think @rdblue recommendation for me is not use immutables, so assuming this comment wont apply anymore?